src/main/java/org/spongepowered/common/command/AndPermissionLevelSubject.java

```diff
@@ -93,4 +96,24 @@ public SubjectData getSubjectData() {
     public MemorySubjectData getTransientSubjectData() {
         return this.opLevelData;
     }
+
+    @Override
+    public boolean hasPermission(Set<Context> contexts, String permission) {
+        final Tristate value = getPermissionValue(contexts, permission);
+        if (value == Tristate.UNDEFINED) {
+            Subject target = this.delegate;
+            while (target instanceof ProxySource) {
+                if (target instanceof SpongeProxySource) {
+                    target = ((SpongeProxySource) target).getSubjectDelegate();
+                } else {
+                    target = ((ProxySource) target).getOriginalSource();
```

Comment:
Point of discussion: I feel like maybe I don't need to, or worse, _shouldn't_ check for this. A `ProxySource` might define its own `hasPermission` et. al. checks - we just know `SpongeProxySource` doesn't. Does it make sense to remove this?